cli/pcluster/config/json_param_types.py

```diff
@@ -251,6 +205,83 @@ def get_default_param_type(self):
         """Get the default Param type managed by the Section type."""
         return JsonParam
 
+    def refresh(self):
+        """Refresh the Json section."""
+        self.refresh_section()
+        super(JsonSection, self).refresh()
+
+    def refresh_section(self):
+        """Perform custom refresh operations."""
+        pass
+
+
+class QueueJsonSection(JsonSection):
+    """JSon Section for queues."""
+
+    def refresh_section(self):
+        """Take values of disable_hyperthreading and enable_efa from cluster section if not specified."""
+        if self.get_param_value("disable_hyperthreading") is None:
+            self.get_param("disable_hyperthreading").value = self.pcluster_config.get_section(
+                "cluster"
+            ).get_param_value("disable_hyperthreading")
+
+        if self.get_param_value("enable_efa") is None:
+            cluster_enable_efa = self.pcluster_config.get_section("cluster").get_param_value("enable_efa")
+            self.get_param("enable_efa").value = cluster_enable_efa == "compute"
+
+        compute_resource_labels = self.get_param_value("compute_resource_settings")
+        if compute_resource_labels:
+            for compute_resource_label in compute_resource_labels.split(","):
+                compute_resource_section = self.pcluster_config.get_section("compute_resource", compute_resource_label)
+                self.refresh_compute_resource(compute_resource_section)
+
+    def refresh_compute_resource(self, compute_resource_section):
+        """
+        Populate additional settings needed for the linked compute resource like vcpus, gpus etc.
+
+        These parameters are set according to queue settings and instance type capabilities.
+        """
+        instance_type_param = compute_resource_section.get_param("instance_type")
+
+        if instance_type_param.value:
+            instance_type = utils.get_instance_type(instance_type_param.value)
+
+            # Set vcpus according to queue's disable_hyperthreading and instance features
+            ht_disabled = self.get_param_value("disable_hyperthreading")
+            vcpus_info = instance_type.get("VCpuInfo")
+            default_cores = vcpus_info.get("DefaultCores")
+            vcpus = default_cores if ht_disabled and default_cores else vcpus_info.get("DefaultVCpus")
+            compute_resource_section.get_param("vcpus").value = vcpus
+
+            # Set gpus according to instance features
+            gpu_info = instance_type.get("GpuInfo", None)
+            if gpu_info:
+                # Currently adding up all gpus. To be reviewed if the case of heterogeneous GPUs arises.
+                compute_resource_section.get_param("gpus").value = sum(
+                    [gpus.get("Count") for gpus in gpu_info.get("Gpus")]
+                )
+
+            # Set enable_efa according to queues' enable_efa and instance features
+            enable_efa = self.get_param_value("enable_efa")
```

Comment:
The problem here is that ``compute_resource`` settings depend on parent ``queue`` settings, which in turn depend on ``cluster`` settings. Moving the logic into single parameters we would not have the call order guarantee.
Additionally, doing everything in a central place allows us to run a single boto3 call, and we do not need a class for each parameter.
About the validation warnings, I can work to move them to validators but this would require another boto3 call as well.

Reply:
Let's  move at least the validator that doesn't require an additional boto3 call and add a comment in the "exceptional" validator we are keeping here, explaining the reason (that is a good one).

Reply:
Is it OK if I just add the comment and we address the proposed refactoring in another patch? I'd like to move both the warnings but we need to find a proper way for handling these cases, and in general to mitigate the exponential growth of boto3 calls.